contrib/windows-cmake/CMakeLists.txt

```diff
@@ -118,53 +193,52 @@ if(NOT HWLOC_SKIP_LSTOPO)
     add_executable(lstopo-no-graphics
         ${LSTOPO_COMMON_SOURCES}
     )
-    target_link_libraries(lstopo-no-graphics PRIVATE libhwloc)
+    target_link_libraries(lstopo-no-graphics PRIVATE hwloc::hwloc)
 
     add_executable(lstopo
         ${LSTOPO_COMMON_SOURCES}
         ${TOPDIR}/utils/lstopo/lstopo-windows.c
     )
-    set_target_properties(lstopo PROPERTIES COMPILE_FLAGS "-DLSTOPO_HAVE_GRAPHICS")
+    target_compile_definitions(lstopo PRIVATE LSTOPO_HAVE_GRAPHICS)
 
-    add_executable(lstopo-win
+    add_executable(lstopo-win WIN32
         ${LSTOPO_COMMON_SOURCES}
         ${TOPDIR}/utils/lstopo/lstopo-windows.c
     )
-    set_target_properties(lstopo-win PROPERTIES COMPILE_FLAGS "-DLSTOPO_HAVE_GRAPHICS")
-    target_link_options(lstopo-win PRIVATE "/subsystem:windows" "/entry:mainCRTStartup")
+    target_compile_definitions(lstopo-win PRIVATE LSTOPO_HAVE_GRAPHICS)
+    target_link_options(lstopo-win PRIVATE "$<$<BOOL:${MSVC}>:/subsystem:windows;/entry:mainCRTStartup>")
```

Comment:
```suggestion
```
outdated?

Reply:
These link flags are still necessary for MSVC as CMake doesn't implicitly add these flags, even with `add_exectuable(lstopo-win WIN32 ...)`

Reply:
the WIN32 option doesn't enable these flags, MSVC needs these additional link flags